publicsuffixlist.js

```diff
@@ -242,13 +242,15 @@ class PublicSuffixList {
 
                 // Ignore surrounding whitespaces
                 line = line.trim();
-                if ( line.length === 0 ) { continue; }
 
-                const exception = line.charCodeAt(0) === 0x21 /* '!' */;
+                const exception = line.length > 0 && line.charCodeAt(0) === 0x21 /* '!' */;
                 if ( exception ) {
                     line = line.slice(1);
                 }
 
+                // https://en.wikipedia.org/wiki/Hostname#Syntax
+                if ( line.length === 0 || line.length > 253 ) { continue; }
```

Comment:
Something to bear in mind here: It's possible to have subdomains made up of emoji.

```js
'👍👍👍👍👍👍👍👍👍👍.example.com'.length // 32
url.domainToASCII('👍👍👍👍👍👍👍👍👍👍.example.com').length // 29
```

The Punycode version of a domain can be _shorter_ than the Unicode version, and in that case valid domains would get rejected here.

Maybe it's best to move this check below after the conversion.

Reply:
>Maybe it's best to move this check below after the conversion.

Done.